### PR TITLE
Support YAML format for Addon advanced configuration values

### PR DESCRIPTION
**Pull request by yuxiang-zhang**
### Description
Closes https://github.com/weaveworks/eksctl-private/issues/779

**Problem**
```
# Valid Configuration in JSON
addons:
- name: coredns
  version: latest
  configurationValues: "{\"replicaCount\":3}"
```
`addons.configurationValues` only accept JSON format. 

```
# Valid in YAML, but should be accepted
addons:
- name: coredns
  version: latest
  configurationValues: |-
    replicaCount: 2
```
Providing the values in such format will result in `Error: replicaCount: 2 is not a valid JSON`. 

**Changes**
Convert YAML to JSON format before calling EKS API. 

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by TiberiuGC — COMMENTED** [review]
I've tested the changes locally, and now it seems like `configurationValues` will accept any (json) string at all as a valid input. e.g.

```yaml
addons:
- name: coredns
  version: latest # auto discovers the latest available
  configurationValues: "not a json"
```

This successfully passes eksctl validations now, but fails when calling EKS API. 

`Yaml schema validation failed with error: [$: string found, object expected]`
  
  The new code seems to successfully convert from yaml to json string, whereas the old code was specifically checking for a json object i.e. 
https://github.com/weaveworks/eksctl/blob/69aee2c12404811cc0a6168a29b313f429ddb1e6/pkg/apis/eksctl.io/v1alpha5/addon.go#L69-L75

I believe that even if the conversion is successful, we still need to check if the result is a valid JSON. Wdyt?

File: userdocs/src/usage/addons.md
Comment (Himangini) [inline userdocs/src/usage/addons.md:137]: there's a typo in line 154 🙈

File: pkg/ctl/create/addon_test.go
Comment (Himangini) [inline pkg/ctl/create/addon_test.go:30]:
```suggestion
			Expect(err).To(MatchError(ContainSubstring(fmt.Sprintf("\"%s\" is not valid, supported format(s) are: JSON and YAML", cfg.Addons[0].ConfigurationValues))))
```

File: pkg/ctl/update/addon_test.go
Comment (Himangini) [inline pkg/ctl/update/addon_test.go:30]:
```suggestion
			Expect(err).To(MatchError(ContainSubstring(fmt.Sprintf("\"%s\" is not valid, supported format(s) are: JSON and YAML", cfg.Addons[0].ConfigurationValues))))
```

File: userdocs/src/usage/addons.md
Comment (Himangini) [inline userdocs/src/usage/addons.md:136]:
```suggestion
`ConfigurationValues` can be provided in the configuration file during the creation or update of addons. Only JSON and YAML formats are supported. 

For eg.,
```

File: userdocs/src/usage/addons.md
Comment (Himangini) [inline userdocs/src/usage/addons.md:156]: ⭐ ❤️

**Review by TiberiuGC — APPROVED** [review]
I have approved this as local tests LGTM after the changes! 👍 
Just make sure we correct the small typo as per review comment and we should be good to merge.

**Comment by yuxiang-zhang** [comment]
> I have approved this as local tests LGTM after the changes! 👍 Just make sure we correct the small typo as per review comment and we should be good to merge.

Thanks for the review and testing! I believe the typo mentioned is already corrected. Kindly inform if it's still there :)

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻 
I'll add a donot-merge label for now. We'll get this merged closer to launch date ✨

**Pull request merged**
